app_dart/lib/src/request_handlers/get_benchmarks.dart

```diff
@@ -30,26 +33,102 @@ class GetBenchmarks extends RequestHandler<Body> {
 
   @override
   Future<Body> get() async {
-    final String branch = request.uri.queryParameters[branchParam] ?? 'master';
-    const int maxRecords = 50;
+    const String master = 'master';
+
+    final int maxRecords = config.maxRecords;
+    final String branch = request.uri.queryParameters[branchParam] ?? master;
     final DatastoreService datastore = datastoreProvider(config.db);
-    final DatastoreDB db = datastore.db;
+    final List<Map<String, dynamic>> benchmarks = <Map<String, dynamic>>[];
+
+    Map<String, Result> releaseBranchMap = <String, Result>{};
+    Map<String, Result> masterMap = <String, Result>{};
+    int masterLimit = maxRecords;
+    int timestamp = DateTime.now().millisecondsSinceEpoch;
+
+    /// Query all commits of the release branch first. Then calcalute the
+    /// number of commits to retrieve from master branch, and obtain
+    /// the starting [timestamp] to filter master commits.
+    if (branch != master) {
+      final List<Commit> releaseBranchCommits = await datastore
+          .queryRecentCommits(limit: maxRecords, branch: branch)
+          .toList();
+      releaseBranchMap = await _getBenchmarks(releaseBranchCommits.length,
+          branch, datastore, releaseBranchCommits, benchmarks, timestamp);
+      masterLimit = maxRecords > releaseBranchCommits.length
+          ? maxRecords - releaseBranchCommits.length
+          : 0;
+      timestamp = releaseBranchCommits.last.timestamp;
+    }
+
+    /// Query all remaining commits from master.
+    if (branch == master || masterLimit > 0) {
```

Comment:
This variable name is really confusing I'd recommend using a different name for it. In this case is not clear if you want 'master' instead.

Reply:
changed to `numberOfMasterCommits` to make it clear.